### PR TITLE
drivers:temperature: Enhancements and corrections for max31865.

### PR DESCRIPTION
**Pull request by jsanbuen**
Added provisions for N-wire,filter, and RC time constant indicators in init_param and device descriptor. Includes formatting corrections.

### CODE REVIEW

**Review by amiclaus — COMMENTED** [review]
please split into 2 commits:
- one for the new additions
- one for the formatting

File: drivers/temperature/max31865pmb1/max31865.c [hunk 1]
```diff
@@ -190,13 +191,13 @@ int max31865_write(struct max31865_dev *device, uint8_t reg_addr,
...
-	if(reg_addr < MAX31865_CONFIG_REG || reg_addr >= MAX31865_FAULTSTAT_REG)
+	if (reg_addr < MAX31865_CONFIG_REG || reg_addr >= MAX31865_FAULTSTAT_REG)
 		return -ENXIO;
 
-	if(reg_addr == MAX31865_RTDMSB_REG || reg_addr == MAX31865_RTDLSB_REG)
+	if (reg_addr == MAX31865_RTDMSB_REG || reg_addr == MAX31865_RTDLSB_REG)
 		return -ENXIO;
 
-	if((reg_addr & MAX31865_WRITE_MASK) != MAX31865_WRITE_MASK)
+	if ((reg_addr & MAX31865_WRITE_MASK) != MAX31865_WRITE_MASK)
```
Comment (CiprianRegus) [inline drivers/temperature/max31865pmb1/max31865.c:200]: Not entirely related to this change, but can't this `if(...)` and just do the or operation anyway?

File: drivers/temperature/max31865pmb1/max31865.c [hunk 2]
```diff
@@ -386,30 +387,30 @@ int max31865_read_rtd(struct max31865_dev *device, uint16_t *rtd_reg)
...
-	if(ret)
+	if (ret)
 		return ret;
 
 	*rtd_reg = *rtd_reg | (uint16_t)reg_data;
 	*rtd_reg >>= 1;
 
 	return max31865_enable_bias(device, false);
-}
+}
```
Comment (CiprianRegus) [inline drivers/temperature/max31865pmb1/max31865.c:416]: add a new line.

File: drivers/temperature/max31865pmb1/max31865.h [hunk 3]
```diff
@@ -136,4 +136,4 @@ int max31865_set_wires(struct max31865_dev *, bool);
 /** Read RTD **/
 int max31865_read_rtd(struct max31865_dev *, uint16_t *);
 
-#endif // __MAX31855_H__
+#endif // __MAX31865_H__
```
Comment (CiprianRegus) [inline drivers/temperature/max31865pmb1/max31865.h:139]: new line here too

**Review by CiprianRegus — APPROVED** [review]
Just a few minor remarks.

File: drivers/temperature/max31865pmb1/max31865.c (hunk 1 above)
Comment (jsanbuen) [reply drivers/temperature/max31865pmb1/max31865.c:200]: Will remove the if statement and just perform the or operation anyway since the first if(...) statement already checks if reg_addr is valid.

**Comment by jsanbuen** [comment]
Edited the enhancement and formatting correction commits for this pull request to address comments.

**Review by amiclaus — COMMENTED** [review]
the commits modifications still seem messed up.

**Comment by jsanbuen** [comment]
Rearranged order of separate commits. Formatting and label correction was done first and additional features were added in succeeding commit.

**Review by amiclaus — COMMENTED** [review]
some more comments.

Please properly write the commit titles (so they do not exceed the maximum character limit).

Also add relevant commit body description for each commit.

File: drivers/temperature/max31865pmb1/max31865.c [hunk 4]
```diff
@@ -138,7 +149,7 @@ int max31865_reg_update(struct max31865_dev *device, uint8_t reg_addr,
  * @brief Read raw register value
  * @param device - MAX31865 descriptor
  * @param reg_addr - register value to read from
- * @param *reg_data - pointer for register value read
```
[the remark this answers is not in the document]
Comment (jsanbuen) [reply drivers/temperature/max31865pmb1/max31865.c:141]: Transferred to formatting corrections commit

**Comment by jsanbuen** [comment]
Addressed comments on formatting change. Sequence still follows original of formatting commit first then enhancements commit.

File: drivers/temperature/max31865pmb1/max31865.c [hunk 5]
```diff
@@ -69,20 +71,29 @@ int max31865_init(struct max31865_dev **device,
 	if (!init_param)
 		return -EINVAL;
 
-	descriptor = (struct max31865_dev *)calloc(1, sizeof(*descriptor));
+	descriptor = (struct max31865_dev *)no_os_calloc(1, sizeof(*descriptor));
```
Comment (amiclaus) [inline drivers/temperature/max31865pmb1/max31865.c:74]: the alloc stuff should go in separate commit.
Also, please follow the proper format for commit title (i.e. `drivers:temperature:max31865pmb: add enhancements`) and the 50/72 commit message rule.

Comment (jsanbuen) [reply drivers/temperature/max31865pmb1/max31865.c:74]: Does this mean that I need to have 3 commits: formatting corrections, enhancements, and no_os_alloc function usage?

Comment (amiclaus) [reply drivers/temperature/max31865pmb1/max31865.c:74]: Yes

**Comment by jsanbuen** [comment]
Separated to three commits: formatting corrections, enhancements, and use of no_os functions for dynamic memory management.

**Review by amiclaus — APPROVED** [review]
looks good!

**Review by amiclaus — COMMENTED** [review]
there are still some conflicts, please make sure you resolve them.

**Comment by jsanbuen** [comment]
Conflicts in specified file addressed through github and marked as resolved.

**Comment by CiprianRegus** [comment]
This cannot be merged. Please manually rebase on master

**Comment by rbolboac** [comment]
@jsanbuen , any updates here? Please see my colleagues request to rebase manually on master so we can proceed with merging. Thank you!

**Review by amiclaus — COMMENTED** [review]
similar to https://github.com/analogdevicesinc/no-OS/pull/1691:
please remove the merge commit. use only rebase or interactive rebase.

**Comment by jsanbuen** [comment]
Resolved conflict via github editor and committed merge (no conflicts with base branch). Separated to two commits: formatting corrections and enhancements. Dynamic memory commit was removed since changes due to inclusion of no_os_alloc were already implemented in the master branch code.

**Comment by CiprianRegus** [comment]
please rebase on master instead of merging.

**Comment by jsanbuen** [comment]
Separated pull request for enhancements (PR #1727 ). This pull request would just be for the formatting corrections for the max31865 driver c and h files since including the commit for enhancements here causes the file conflict with master.

**Review by amiclaus — APPROVED** [review]

**Review by amiclaus — COMMENTED** [review]
after the merg of #1727 some conflicts appeared. please address them. (this might be avoided in the future if you keep separate commits, but in the same PR)

**Comment by jsanbuen** [comment]
> after the merg of #1727 some conflicts appeared. please address them. (this might be avoided in the future if you keep separate commits, but in the same PR)

Addressed conflicts by using currently merged in master version and applying the formatting corrections on that version. For reference of future pull requests with multiple commits, does it mean that each commit should have the changes applied only to the currently in master and not having the changes cascaded or applied after every commit like what I initially did?

**Review by amiclaus — APPROVED** [review]

**Comment by amiclaus** [comment]
> > after the merg of #1727 some conflicts appeared. please address them. (this might be avoided in the future if you keep separate commits, but in the same PR)
> 
> Addressed conflicts by using currently merged in master version and applying the formatting corrections on that version. For reference of future pull requests with multiple commits, does it mean that each commit should have the changes applied only to the currently in master and not having the changes cascaded or applied after every commit like what I initially did?

You can cascade commits in the same PR as you did initially, but those commits were mixed up (for example the formatting commit contained also some of the enhancements and vice versa). As long as the commits are properly adjusted and do what the commit title/description says, you can add them in the same PR (if they are related to the same part/project)
